### PR TITLE
weevely: init at 4.0.2-unstable-2024-04-29

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/we/weevely/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+, fetchFromGitHub
+, python3
+, python3Packages
+, makeWrapper
+, installShellFiles
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "weevely";
+  version = "4.0.1-unstable-2023-06-18";
```
Comment (felbinger) [inline pkgs/by-name/we/weevely/package.nix:11]: Can't see at all where this `unstable` and date is coming from. I think it should be just `4.0.1`

File: pkgs/by-name/we/weevely/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "epinna";
+    repo = "weevely3";
+    rev = "445bd88be1a7ae2a1aaa34b88a38a92786e56b84";
```
Comment (felbinger) [inline pkgs/by-name/we/weevely/package.nix:17]: Maybe you should use the git tag `v4.0.2`, which makes it easier to read, see: https://github.com/epinna/weevely3/tree/v4.0.2

The commit rev you put in is neither version 4.0.1 nor 4.0.2, it's something in between them...

File: pkgs/by-name/we/weevely/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+, python3
+, python3Packages
+, makeWrapper
+, installShellFiles
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "weevely";
+  version = "4.0.1-unstable-2023-06-18";
+  format = "other";
```
Comment (felbinger) [inline pkgs/by-name/we/weevely/package.nix:12]: As far as I know, format is deprecated (can't prove it right now), consider using `pyproject = false;`, see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/by-name/we/weevely/package.nix (hunk 1 above)
Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix:11]: I will use 4.0.2

File: pkgs/by-name/we/weevely/package.nix (hunk 3 above)
Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix:12]: Done

File: pkgs/by-name/we/weevely/package.nix (hunk 2 above)
Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix:17]: I will use the latest commit.

File: pkgs/by-name/we/weevely/package.nix (hunk 1 above)
Comment (felbinger) [reply pkgs/by-name/we/weevely/package.nix]: But why should you package an unstable version? Wouldn't it make more sense to use the latest version?

Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix]: > But why should you package an unstable version? Wouldn't it make more sense to use the latest version?

Read the latest commit. It is a good fiix to have.

File: pkgs/by-name/we/weevely/package.nix (hunk 2 above)
Comment (felbinger) [reply pkgs/by-name/we/weevely/package.nix]: If you really need this version (which is 4 commit ahead of the last release), you should maybe ask epinna to create a new relase...

Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix]: For just 4 commits, it is not needed to create a new release.

File: pkgs/by-name/we/weevely/package.nix (hunk 1 above)
Comment (felbinger) [reply pkgs/by-name/we/weevely/package.nix]: In my opinion you should ask epinna to create a new release then. Where should this end? Creating a new package update pull request every single time he commits anything to this project?

Comment (D3vil0p3r) [reply pkgs/by-name/we/weevely/package.nix]: Imho there is no need. But if you would like to ask you to the dev, and it accepts, I will update it to 4.0.3.

**Review by felbinger — CHANGES_REQUESTED** [review]
Use release instead of latest commits

File: pkgs/by-name/we/weevely/package.nix (hunk 1 above)
Comment (felbinger) [reply pkgs/by-name/we/weevely/package.nix]: To then use the latest commit again, when he pushes some new commits? Don't see the point

File: pkgs/by-name/we/weevely/package.nix (hunk 2 above)
Comment (felbinger) [reply pkgs/by-name/we/weevely/package.nix]: Then I guess the features aren't that important :)

**Pull request merged**